{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genome assembly"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "                  __    __    __    __\n",
      "                 /  \\  /  \\  /  \\  /  \\\n",
      "                /    \\/    \\/    \\/    \\\n",
      "███████████████/  /██/  /██/  /██/  /████████████████████████\n",
      "              /  / \\   / \\   / \\   / \\  \\____\n",
      "             /  /   \\_/   \\_/   \\_/   \\    o \\__,\n",
      "            / _/                       \\_____/  `\n",
      "            |/\n",
      "        ███╗   ███╗ █████╗ ███╗   ███╗██████╗  █████╗\n",
      "        ████╗ ████║██╔══██╗████╗ ████║██╔══██╗██╔══██╗\n",
      "        ██╔████╔██║███████║██╔████╔██║██████╔╝███████║\n",
      "        ██║╚██╔╝██║██╔══██║██║╚██╔╝██║██╔══██╗██╔══██║\n",
      "        ██║ ╚═╝ ██║██║  ██║██║ ╚═╝ ██║██████╔╝██║  ██║\n",
      "        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝\n",
      "\n",
      "        mamba (0.27.0) supported by @QuantStack\n",
      "\n",
      "        GitHub:  https://github.com/mamba-org/mamba\n",
      "        Twitter: https://twitter.com/QuantStack\n",
      "\n",
      "█████████████████████████████████████████████████████████████\n",
      "\n",
      "\n",
      "Looking for: ['fastqc', 'fastp', 'kraken2', 'spades', 'bowtie2', 'blast']\n",
      "\n",
      "bioconda/linux-64                                           Using cache\n",
      "bioconda/noarch                                             Using cache\n",
      "conda-forge/linux-64                                        Using cache\n",
      "conda-forge/noarch                                          Using cache\n",
      "Transaction\n",
      "\n",
      "  Prefix: /workspace/Exercises-ECV/packages/dev\n",
      "\n",
      "  Updating specs:\n",
      "\n",
      "   - fastqc\n",
      "   - fastp\n",
      "   - kraken2\n",
      "   - spades\n",
      "   - bowtie2\n",
      "   - blast\n",
      "\n",
      "\n",
      "  Package                                      Version  Build               Channel                    Size\n",
      "─────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
      "  Install:\n",
      "─────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
      "\n",
      "\u001b[32m  + _libgcc_mutex                       \u001b[00m           0.1  conda_forge         conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + _openmp_mutex                       \u001b[00m           4.5  2_gnu               conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + _sysroot_linux-64_curr_repodata_hack\u001b[00m             3  h5bd9786_13         conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + alsa-lib                            \u001b[00m         1.2.8  h166bdaf_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + blast                               \u001b[00m        2.13.0  hf3cf87c_0          bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + bowtie2                             \u001b[00m         2.5.1  py310h8d7afc0_0     bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + bzip2                               \u001b[00m         1.0.8  h7f98852_4          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + c-ares                              \u001b[00m        1.18.1  h7f98852_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + ca-certificates                     \u001b[00m     2022.12.7  ha878542_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + cairo                               \u001b[00m        1.16.0  h35add3b_1015       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + curl                                \u001b[00m        7.88.1  hdc1c0ab_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + entrez-direct                       \u001b[00m          16.2  he881be0_1          bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + expat                               \u001b[00m         2.5.0  h27087fc_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + fastp                               \u001b[00m        0.23.2  h5f740d0_3          bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + fastqc                              \u001b[00m        0.12.1  hdfd78af_0          bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + font-ttf-dejavu-sans-mono           \u001b[00m          2.37  hab24e00_0          conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + font-ttf-inconsolata                \u001b[00m         3.000  h77eed37_0          conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + font-ttf-source-code-pro            \u001b[00m         2.038  h77eed37_0          conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + font-ttf-ubuntu                     \u001b[00m          0.83  hab24e00_0          conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + fontconfig                          \u001b[00m        2.14.2  h14ed4e7_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + fonts-conda-ecosystem               \u001b[00m             1  0                   conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + fonts-conda-forge                   \u001b[00m             1  0                   conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + freetype                            \u001b[00m        2.12.1  hca18f0e_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + gettext                             \u001b[00m        0.21.1  h27087fc_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + giflib                              \u001b[00m         5.2.1  h0b41bf4_3          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + graphite2                           \u001b[00m        1.3.13  h58526e2_1001       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + harfbuzz                            \u001b[00m         6.0.0  h3ff4399_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + icu                                 \u001b[00m          72.1  hcb278e6_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + isa-l                               \u001b[00m        2.30.0  ha770c72_4          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + jpeg                                \u001b[00m            9e  h0b41bf4_3          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + kernel-headers_linux-64             \u001b[00m        3.10.0  h4a8ded7_13         conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + keyutils                            \u001b[00m         1.6.1  h166bdaf_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + kraken2                             \u001b[00m         2.1.2  pl5321h9f5acd7_3    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + krb5                                \u001b[00m        1.20.1  h81ceb04_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + lcms2                               \u001b[00m          2.14  h6ed2654_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + ld_impl_linux-64                    \u001b[00m          2.40  h41732ed_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + lerc                                \u001b[00m         4.0.0  h27087fc_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libcups                             \u001b[00m         2.3.3  h36d4200_3          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libcurl                             \u001b[00m        7.88.1  hdc1c0ab_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libdeflate                          \u001b[00m          1.13  h166bdaf_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libedit                             \u001b[00m  3.1.20191231  he28a2e2_2          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libev                               \u001b[00m          4.33  h516909a_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libffi                              \u001b[00m         3.4.2  h7f98852_5          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libgcc-ng                           \u001b[00m        12.2.0  h65d4601_19         conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libglib                             \u001b[00m        2.74.1  h606061b_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libgomp                             \u001b[00m        12.2.0  h65d4601_19         conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libhwloc                            \u001b[00m         2.9.0  hd6dc26d_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libiconv                            \u001b[00m          1.17  h166bdaf_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libidn2                             \u001b[00m         2.3.4  h166bdaf_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libnghttp2                          \u001b[00m        1.52.0  h61bc06f_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libnsl                              \u001b[00m         2.0.0  h7f98852_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libpng                              \u001b[00m        1.6.39  h753d276_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libsqlite                           \u001b[00m        3.40.0  h753d276_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libssh2                             \u001b[00m        1.10.0  hf14f497_3          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libstdcxx-ng                        \u001b[00m        12.2.0  h46fd767_19         conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libtiff                             \u001b[00m         4.4.0  h0e0dad5_3          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libunistring                        \u001b[00m        0.9.10  h7f98852_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libuuid                             \u001b[00m        2.32.1  h7f98852_1000       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libwebp-base                        \u001b[00m         1.3.0  h0b41bf4_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libxcb                              \u001b[00m          1.13  h7f98852_1004       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libxml2                             \u001b[00m        2.10.3  hfdac1af_6          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + libzlib                             \u001b[00m        1.2.13  h166bdaf_4          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + llvm-openmp                         \u001b[00m         8.0.1  hc9558a2_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + lz4-c                               \u001b[00m         1.9.4  hcb278e6_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + ncurses                             \u001b[00m           6.3  h27087fc_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + openjdk                             \u001b[00m        17.0.3  h58dac75_5          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + openmp                              \u001b[00m         8.0.1  0                   conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + openssl                             \u001b[00m         3.1.0  h0b41bf4_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + pcre                                \u001b[00m          8.45  h9c3ff4c_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + pcre2                               \u001b[00m         10.40  hc3806b6_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl                                \u001b[00m        5.32.1  2_h7f98852_perl5    conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-archive-tar                    \u001b[00m          2.40  pl5321hdfd78af_0    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-carp                           \u001b[00m          1.38  pl5321hdfd78af_4    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-common-sense                   \u001b[00m          3.75  pl5321hdfd78af_0    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-compress-raw-bzip2             \u001b[00m         2.201  pl5321h87f3376_1    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-compress-raw-zlib              \u001b[00m         2.105  pl5321h87f3376_0    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-encode                         \u001b[00m          3.19  pl5321hec16e2b_1    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-exporter                       \u001b[00m          5.72  pl5321hdfd78af_2    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-exporter-tiny                  \u001b[00m      1.002002  pl5321hdfd78af_0    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-extutils-makemaker             \u001b[00m          7.70  pl5321hd8ed1ab_0    conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-io-compress                    \u001b[00m         2.201  pl5321h87f3376_0    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-io-zlib                        \u001b[00m          1.14  pl5321hdfd78af_0    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-json                           \u001b[00m          4.10  pl5321hdfd78af_0    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-json-xs                        \u001b[00m          2.34  pl5321h9f5acd7_5    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-list-moreutils                 \u001b[00m         0.430  pl5321hdfd78af_0    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-list-moreutils-xs              \u001b[00m         0.430  pl5321hec16e2b_1    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-parent                         \u001b[00m         0.236  pl5321hdfd78af_2    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-pathtools                      \u001b[00m          3.75  pl5321hec16e2b_3    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-scalar-list-utils              \u001b[00m          1.62  pl5321hec16e2b_1    bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + perl-types-serialiser               \u001b[00m          1.01  pl5321hdfd78af_0    bioconda/noarch     \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + pip                                 \u001b[00m        23.0.1  pyhd8ed1ab_0        conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + pixman                              \u001b[00m        0.40.0  h36c2ea0_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + popt                                \u001b[00m          1.16  1                   bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + pthread-stubs                       \u001b[00m           0.4  h36c2ea0_1001       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + python                              \u001b[00m       3.10.10  he550d4f_0_cpython  conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + python_abi                          \u001b[00m          3.10  3_cp310             conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + readline                            \u001b[00m           8.2  h8228510_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + rsync                               \u001b[00m         3.2.7  h70740c4_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + setuptools                          \u001b[00m        67.6.1  pyhd8ed1ab_0        conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + spades                              \u001b[00m        3.15.5  h95f258a_1          bioconda/linux-64   \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + sysroot_linux-64                    \u001b[00m          2.17  h4a8ded7_13         conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + tar                                 \u001b[00m          1.34  hb2e2bae_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + tbb                                 \u001b[00m      2021.8.0  hf52228f_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + tk                                  \u001b[00m        8.6.12  h27826a3_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + tzdata                              \u001b[00m         2023c  h71feb2d_0          conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + wget                                \u001b[00m        1.20.3  ha35d2d1_1          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + wheel                               \u001b[00m        0.40.0  pyhd8ed1ab_0        conda-forge/noarch  \u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-fixesproto                     \u001b[00m           5.0  h7f98852_1002       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-inputproto                     \u001b[00m         2.3.2  h7f98852_1002       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-kbproto                        \u001b[00m         1.0.7  h7f98852_1002       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libice                         \u001b[00m        1.0.10  h7f98852_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libsm                          \u001b[00m         1.2.3  hd9c2040_1000       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libx11                         \u001b[00m         1.8.4  h0b41bf4_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libxau                         \u001b[00m         1.0.9  h7f98852_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libxdmcp                       \u001b[00m         1.1.3  h7f98852_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libxext                        \u001b[00m         1.3.4  h0b41bf4_2          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libxfixes                      \u001b[00m         5.0.3  h7f98852_1004       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libxi                          \u001b[00m        1.7.10  h7f98852_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libxrender                     \u001b[00m        0.9.10  h7f98852_1003       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-libxtst                        \u001b[00m         1.2.3  h7f98852_1002       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-recordproto                    \u001b[00m        1.14.2  h7f98852_1002       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-renderproto                    \u001b[00m        0.11.1  h7f98852_1002       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-xextproto                      \u001b[00m         7.3.0  h0b41bf4_1003       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xorg-xproto                         \u001b[00m        7.0.31  h7f98852_1007       conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xxhash                              \u001b[00m         0.8.0  h7f98852_3          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + xz                                  \u001b[00m         5.2.6  h166bdaf_0          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + zlib                                \u001b[00m        1.2.13  h166bdaf_4          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\u001b[32m  + zstd                                \u001b[00m         1.5.2  h3eb15da_6          conda-forge/linux-64\u001b[32m     Cached\u001b[00m\n",
      "\n",
      "  Summary:\n",
      "\n",
      "  Install: 128 packages\n",
      "\n",
      "  Total download: 0 B\n",
      "\n",
      "─────────────────────────────────────────────────────────────────────────────────────────────────────────────\n",
      "\n",
      "\u001b[?25l\u001b[2K\u001b[0G\u001b[?25hPreparing transaction: done\n",
      "Verifying transaction: done\n",
      "Executing transaction: done\n",
      "\n",
      "To activate this environment, use\n",
      "\n",
      "     $ mamba activate dev\n",
      "\n",
      "To deactivate an active environment, use\n",
      "\n",
      "     $ mamba deactivate\n",
      "\n"
     ]
    }
   ],
   "source": [
    "mamba create -n dev -y -c bioconda fastqc fastp kraken2 spades bowtie2 blast\n",
    "conda activate dev"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "usage: fastp [options] ... \n",
      "options:\n",
      "  -i, --in1                            read1 input file name (string [=])\n",
      "  -o, --out1                           read1 output file name (string [=])\n",
      "  -I, --in2                            read2 input file name (string [=])\n",
      "  -O, --out2                           read2 output file name (string [=])\n",
      "      --unpaired1                      for PE input, if read1 passed QC but read2 not, it will be written to unpaired1. Default is to discard it. (string [=])\n",
      "      --unpaired2                      for PE input, if read2 passed QC but read1 not, it will be written to unpaired2. If --unpaired2 is same as --unpaired1 (default mode), both unpaired reads will be written to this same file. (string [=])\n",
      "      --overlapped_out                 for each read pair, output the overlapped region if it has no any mismatched base. (string [=])\n",
      "      --failed_out                     specify the file to store reads that cannot pass the filters. (string [=])\n",
      "  -m, --merge                          for paired-end input, merge each pair of reads into a single read if they are overlapped. The merged reads will be written to the file given by --merged_out, the unmerged reads will be written to the files specified by --out1 and --out2. The merging mode is disabled by default.\n",
      "      --merged_out                     in the merging mode, specify the file name to store merged output, or specify --stdout to stream the merged output (string [=])\n",
      "      --include_unmerged               in the merging mode, write the unmerged or unpaired reads to the file specified by --merge. Disabled by default.\n",
      "  -6, --phred64                        indicate the input is using phred64 scoring (it'll be converted to phred33, so the output will still be phred33)\n",
      "  -z, --compression                    compression level for gzip output (1 ~ 9). 1 is fastest, 9 is smallest, default is 4. (int [=4])\n",
      "      --stdin                          input from STDIN. If the STDIN is interleaved paired-end FASTQ, please also add --interleaved_in.\n",
      "      --stdout                         stream passing-filters reads to STDOUT. This option will result in interleaved FASTQ output for paired-end output. Disabled by default.\n",
      "      --interleaved_in                 indicate that <in1> is an interleaved FASTQ which contains both read1 and read2. Disabled by default.\n",
      "      --reads_to_process               specify how many reads/pairs to be processed. Default 0 means process all reads. (int [=0])\n",
      "      --dont_overwrite                 don't overwrite existing files. Overwritting is allowed by default.\n",
      "      --fix_mgi_id                     the MGI FASTQ ID format is not compatible with many BAM operation tools, enable this option to fix it.\n",
      "  -V, --verbose                        output verbose log information (i.e. when every 1M reads are processed).\n",
      "  -A, --disable_adapter_trimming       adapter trimming is enabled by default. If this option is specified, adapter trimming is disabled\n",
      "  -a, --adapter_sequence               the adapter for read1. For SE data, if not specified, the adapter will be auto-detected. For PE data, this is used if R1/R2 are found not overlapped. (string [=auto])\n",
      "      --adapter_sequence_r2            the adapter for read2 (PE data only). This is used if R1/R2 are found not overlapped. If not specified, it will be the same as <adapter_sequence> (string [=auto])\n",
      "      --adapter_fasta                  specify a FASTA file to trim both read1 and read2 (if PE) by all the sequences in this FASTA file (string [=])\n",
      "      --detect_adapter_for_pe          by default, the auto-detection for adapter is for SE data input only, turn on this option to enable it for PE data.\n",
      "  -f, --trim_front1                    trimming how many bases in front for read1, default is 0 (int [=0])\n",
      "  -t, --trim_tail1                     trimming how many bases in tail for read1, default is 0 (int [=0])\n",
      "  -b, --max_len1                       if read1 is longer than max_len1, then trim read1 at its tail to make it as long as max_len1. Default 0 means no limitation (int [=0])\n",
      "  -F, --trim_front2                    trimming how many bases in front for read2. If it's not specified, it will follow read1's settings (int [=0])\n",
      "  -T, --trim_tail2                     trimming how many bases in tail for read2. If it's not specified, it will follow read1's settings (int [=0])\n",
      "  -B, --max_len2                       if read2 is longer than max_len2, then trim read2 at its tail to make it as long as max_len2. Default 0 means no limitation. If it's not specified, it will follow read1's settings (int [=0])\n",
      "  -D, --dedup                          enable deduplication to drop the duplicated reads/pairs\n",
      "      --dup_calc_accuracy              accuracy level to calculate duplication (1~6), higher level uses more memory (1G, 2G, 4G, 8G, 16G, 24G). Default 1 for no-dedup mode, and 3 for dedup mode. (int [=0])\n",
      "      --dont_eval_duplication          don't evaluate duplication rate to save time and use less memory.\n",
      "  -g, --trim_poly_g                    force polyG tail trimming, by default trimming is automatically enabled for Illumina NextSeq/NovaSeq data\n",
      "      --poly_g_min_len                 the minimum length to detect polyG in the read tail. 10 by default. (int [=10])\n",
      "  -G, --disable_trim_poly_g            disable polyG tail trimming, by default trimming is automatically enabled for Illumina NextSeq/NovaSeq data\n",
      "  -x, --trim_poly_x                    enable polyX trimming in 3' ends.\n",
      "      --poly_x_min_len                 the minimum length to detect polyX in the read tail. 10 by default. (int [=10])\n",
      "  -5, --cut_front                      move a sliding window from front (5') to tail, drop the bases in the window if its mean quality < threshold, stop otherwise.\n",
      "  -3, --cut_tail                       move a sliding window from tail (3') to front, drop the bases in the window if its mean quality < threshold, stop otherwise.\n",
      "  -r, --cut_right                      move a sliding window from front to tail, if meet one window with mean quality < threshold, drop the bases in the window and the right part, and then stop.\n",
      "  -W, --cut_window_size                the window size option shared by cut_front, cut_tail or cut_sliding. Range: 1~1000, default: 4 (int [=4])\n",
      "  -M, --cut_mean_quality               the mean quality requirement option shared by cut_front, cut_tail or cut_sliding. Range: 1~36 default: 20 (Q20) (int [=20])\n",
      "      --cut_front_window_size          the window size option of cut_front, default to cut_window_size if not specified (int [=4])\n",
      "      --cut_front_mean_quality         the mean quality requirement option for cut_front, default to cut_mean_quality if not specified (int [=20])\n",
      "      --cut_tail_window_size           the window size option of cut_tail, default to cut_window_size if not specified (int [=4])\n",
      "      --cut_tail_mean_quality          the mean quality requirement option for cut_tail, default to cut_mean_quality if not specified (int [=20])\n",
      "      --cut_right_window_size          the window size option of cut_right, default to cut_window_size if not specified (int [=4])\n",
      "      --cut_right_mean_quality         the mean quality requirement option for cut_right, default to cut_mean_quality if not specified (int [=20])\n",
      "  -Q, --disable_quality_filtering      quality filtering is enabled by default. If this option is specified, quality filtering is disabled\n",
      "  -q, --qualified_quality_phred        the quality value that a base is qualified. Default 15 means phred quality >=Q15 is qualified. (int [=15])\n",
      "  -u, --unqualified_percent_limit      how many percents of bases are allowed to be unqualified (0~100). Default 40 means 40% (int [=40])\n",
      "  -n, --n_base_limit                   if one read's number of N base is >n_base_limit, then this read/pair is discarded. Default is 5 (int [=5])\n",
      "  -e, --average_qual                   if one read's average quality score <avg_qual, then this read/pair is discarded. Default 0 means no requirement (int [=0])\n",
      "  -L, --disable_length_filtering       length filtering is enabled by default. If this option is specified, length filtering is disabled\n",
      "  -l, --length_required                reads shorter than length_required will be discarded, default is 15. (int [=15])\n",
      "      --length_limit                   reads longer than length_limit will be discarded, default 0 means no limitation. (int [=0])\n",
      "  -y, --low_complexity_filter          enable low complexity filter. The complexity is defined as the percentage of base that is different from its next base (base[i] != base[i+1]).\n",
      "  -Y, --complexity_threshold           the threshold for low complexity filter (0~100). Default is 30, which means 30% complexity is required. (int [=30])\n",
      "      --filter_by_index1               specify a file contains a list of barcodes of index1 to be filtered out, one barcode per line (string [=])\n",
      "      --filter_by_index2               specify a file contains a list of barcodes of index2 to be filtered out, one barcode per line (string [=])\n",
      "      --filter_by_index_threshold      the allowed difference of index barcode for index filtering, default 0 means completely identical. (int [=0])\n",
      "  -c, --correction                     enable base correction in overlapped regions (only for PE data), default is disabled\n",
      "      --overlap_len_require            the minimum length to detect overlapped region of PE reads. This will affect overlap analysis based PE merge, adapter trimming and correction. 30 by default. (int [=30])\n",
      "      --overlap_diff_limit             the maximum number of mismatched bases to detect overlapped region of PE reads. This will affect overlap analysis based PE merge, adapter trimming and correction. 5 by default. (int [=5])\n",
      "      --overlap_diff_percent_limit     the maximum percentage of mismatched bases to detect overlapped region of PE reads. This will affect overlap analysis based PE merge, adapter trimming and correction. Default 20 means 20%. (int [=20])\n",
      "  -U, --umi                            enable unique molecular identifier (UMI) preprocessing\n",
      "      --umi_loc                        specify the location of UMI, can be (index1/index2/read1/read2/per_index/per_read, default is none (string [=])\n",
      "      --umi_len                        if the UMI is in read1/read2, its length should be provided (int [=0])\n",
      "      --umi_prefix                     if specified, an underline will be used to connect prefix and UMI (i.e. prefix=UMI, UMI=AATTCG, final=UMI_AATTCG). No prefix by default (string [=])\n",
      "      --umi_skip                       if the UMI is in read1/read2, fastp can skip several bases following UMI, default is 0 (int [=0])\n",
      "  -p, --overrepresentation_analysis    enable overrepresented sequence analysis.\n",
      "  -P, --overrepresentation_sampling    one in (--overrepresentation_sampling) reads will be computed for overrepresentation analysis (1~10000), smaller is slower, default is 20. (int [=20])\n",
      "  -j, --json                           the json format report file name (string [=fastp.json])\n",
      "  -h, --html                           the html format report file name (string [=fastp.html])\n",
      "  -R, --report_title                   should be quoted with ' or \", default is \"fastp report\" (string [=fastp report])\n",
      "  -w, --thread                         worker thread number, default is 3 (int [=3])\n",
      "  -s, --split                          split output by limiting total split file number with this option (2~999), a sequential number prefix will be added to output name ( 0001.out.fq, 0002.out.fq...), disabled by default (int [=0])\n",
      "  -S, --split_by_lines                 split output by limiting lines of each file with this option(>=1000), a sequential number prefix will be added to output name ( 0001.out.fq, 0002.out.fq...), disabled by default (long [=0])\n",
      "  -d, --split_prefix_digits            the digits for the sequential number padding (1~10), default is 4, so the filename will be padded as 0001.xxx, 0 to disable padding (int [=4])\n",
      "      --cut_by_quality5                DEPRECATED, use --cut_front instead.\n",
      "      --cut_by_quality3                DEPRECATED, use --cut_tail instead.\n",
      "      --cut_by_quality_aggressive      DEPRECATED, use --cut_right instead.\n",
      "      --discard_unmerged               DEPRECATED, no effect now, see the introduction for merging.\n",
      "  -?, --help                           print this message\n",
      "(/workspace/Exercises-ECV/packages/dev) \n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "fastp --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hello\n"
     ]
    }
   ],
   "source": [
    "echo \"hello\""
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Download the kraken2 DB: "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "mkdir -p databases/kraken2; cd databases/kraken2\n",
    "wget https://genome-idx.s3.amazonaws.com/kraken/k2_viral_20230314.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [],
   "source": [
    "tar -xvf k2_viral_20230314.tar.gz; rm test.tar.gz\n",
    "\n",
    "cd ../../"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
